pkg/server/endpoints/authorized_entryfetcher_attested_nodes.go

```diff
@@ -33,7 +36,7 @@ type attestedNodes struct {
 
 // buildAttestedNodesCache fetches all attested nodes and adds the unexpired ones to the cache.
 // It runs once at startup.
-func buildAttestedNodesCache(ctx context.Context, log logrus.FieldLogger, ds datastore.DataStore, clk clock.Clock, cache *authorizedentries.Cache, sqlTransactionTimeout time.Duration) (*attestedNodes, error) {
+func buildAttestedNodesCache(ctx context.Context, log logrus.FieldLogger, metrics telemetry.Metrics, ds datastore.DataStore, clk clock.Clock, cache *authorizedentries.Cache, sqlTransactionTimeout time.Duration) (*attestedNodes, error) {
```

Comment:
Pre-existing condition (don't address here): we should consider wrapping the large number of args to this function into an arg struct. None are "plain" types like string so it's not really an issue per se, but 7 args is a lot. Similarly for `buildCache` in `authorized_entryfetcher.go`.

Reply:
I agree. I almost did in this PR but opted against as the PR was already getting long and I wanted to maintain the metrics-centered context.